### PR TITLE
config,config-loader: add support for suffixed config files and multiple roots

### PR DESCRIPTION
**Pull request by Rugvip**
Enables things like `app-config.local.yaml`, `packages/app/app-config.development.yaml`

### CODE REVIEW

**Comment by andrewthauer** [comment]
👍 I think this will help workaround an issue I ran into where environment variables (e.g. `APP_CONFIG_backend_baseUrl`) were being upper when deployed. Which obviously did not work as desired. My current hack was to use `jq` with a replacement map.

Adding an `app-config.production.yaml` file should help in this particular case. Do these act as overrides similar to how a `docker-compose.override.yml` file would by inheriting values?

**Comment by Rugvip** [comment]
@andrewthauer 👍 

Merging works at least similar to docker-compose. Iirc they have some custom logic for merging certain things like ports and volumes. We don't try to merge arrays together at all and just let them override each other.

I documented the file resolution and merging rules here: https://github.com/spotify/backstage/blob/2af561308b137c60e57c35bc8948d6febe295056/docs/conf/writing.md#file-resolution

**Comment by soapraj** [comment]
+1 for adding support to `packages/a/app-config.yaml`

- Want to understand a use-case for `app-config.production.yaml` and `app-config.production.local.yaml`? 

- In this implementation app-config.yaml in the root would get the least priority during resolution if you any other config file with a .local to .{env} suffix.

Should we replace app-config.yaml with app-config.production.yaml at the root and always need the config files to be tied to a specific env? To me, it sounds confusing to keep one more config file without any suffix.

File: packages/config-loader/src/lib/resolver.ts [hunk 1]
```diff
@@ -15,21 +15,45 @@
...
   options: ResolveOptions,
 ): Promise<string[]> {
-  // TODO: We'll want this to be a bit more elaborate, probably adding configs for
-  //       specific env, and maybe local config for plugins.
-  const configPath = resolvePath(options.rootPath, 'app-config.yaml');
+  const filePaths = [
+    `app-config.yaml`,
+    `app-config.local.yaml`,
+    `app-config.${options.env}.yaml`,
+    `app-config.${options.env}.local.yaml`,
```
Comment (freben) [inline packages/config-loader/src/lib/resolver.ts:44]: hm not sure if needed - like, would you use a testing.local for example?

**Review by freben — APPROVED** [review]

**Comment by Rugvip** [comment]
@soapraj Use-case for env files is to be able to have separate config for different envs without a lot of duplication. Usually you just wanna switch out a few things between different envs.

The use-case for `.local` is that it's `.gitignore`'d and you can keep some secrets and whatnot there for development.

File: packages/config-loader/src/lib/resolver.ts (hunk 1 above)
Comment (Rugvip) [reply packages/config-loader/src/lib/resolver.ts:44]: Bit just for completeness, but it is needed if you want to override things in an env config, as the env config has higher prio than local. Could ofc consider changing that too though :grin:

**Pull request merged**